skimage/filters/rank/tests/test_rank.py

```diff
@@ -114,16 +125,30 @@ def test_rank_filter_selem_kwarg_deprecation(self, filter):
             getattr(rank, filter)(self.image.astype(np.uint8),
                                   selem=self.footprint)
 
+    @parametrize('outdt', [None, np.float32, np.float64])
     @parametrize('filter', ['equalize', 'otsu', 'autolevel', 'gradient',
                             'majority', 'maximum', 'mean', 'geometric_mean',
                             'subtract_mean', 'median', 'minimum', 'modal',
                             'enhance_contrast', 'pop', 'sum', 'threshold',
                             'noise_filter', 'entropy'])
-    def test_rank_filters_3D(self, filter):
+    def test_rank_filters_3D(self, filter, outdt):
         @test_parallel(warnings_matching=['Possible precision loss'])
         def check():
             expected = self.refs_3d[filter]
-            result = getattr(rank, filter)(self.volume, self.footprint_3d)
+            if outdt is not None:
+                out = np.zeros_like(expected, dtype=outdt)
+            else:
+                out = None
+            result = getattr(rank, filter)(self.volume, self.footprint_3d,
+                                           out=out)
+            if outdt is not None:
+                # Avoid rounding issues comparing to expected result
+                if filter == 'sum':
+                    # sum test data seems to be 8-bit disguised as 16-bit
+                    datadt = np.uint8
+                else:
+                    datadt = expected.dtype
+                result = result.astype(datadt)
```

Comment:
```suggestion
                result = result.astype(expected.dtype)
```
Isn't @grlee77's suggestion applicable here also?

Reply:
Only if the test data for the 3D case was fixed to be uint8 instead of uint16, which IIUC requires a PR on gitlab.